src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -218,6 +227,14 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
 
         case "$T":
           TypeName typeName = (TypeName) codeBlock.args.get(a++);
+          if (typeName instanceof ClassName) {
+            ClassName candidate = (ClassName) typeName;
+            if (staticImportClassNames.contains(candidate.canonicalName)) {
+              checkState(potentiallyStaticImported == null, "pending type for static import?!");
```

Comment:
It actually throws for anything that looks like: `$T*$T` with `*` is any number of non-default parts:

    addStatement("$1T$2L$1T", Runtime.class, "?") // illegal if $T is static imported